wagtail/fields.py

```diff
@@ -22,6 +22,7 @@ def __init__(self, *args, **kwargs):
         # and retrospectively adding them would generate unwanted migration noise
         self.editor = kwargs.pop("editor", "default")
         self.features = kwargs.pop("features", None)
+        self.search_index = True
```

Comment:
The changes here are not needed - RichTextField is a distinct thing from StreamField, and we're not providing a `search_index` option there (if you wanted to exclude a rich text field from search indexing, you'd just leave it out of `search_fields`).